xcode/iconizer.sh

```diff
@@ -1,52 +1,78 @@
 #!/bin/bash
 set -euo pipefail
 
-# Iconizer shell script by Stefan Herold
+# Iconizer shell script by Stefan Herold and Yunus Tür
 
 # This is a simple tool to generate all necessary app icon sizes from one vector file.
 # To use: specify the path to your vector graphic (PDF format)
-# Example: sh iconizer.sh MyVectorGraphic.pdf
+# Example: sh iconizer.sh VectorGraphic.pdf AppIcon.appiconset
 
 # Requires ImageMagick: http://www.imagemagick.org/
 # Requires jq: https://stedolan.github.io/jq/
 
-CONTENTS_FILE="Contents.json"
+# Variables
 
-die () {
-  echo >&2 "$@"
-  exit 1
-}
+PDF_PATH=""
+APP_ICON_SET_PATH=""
+CONTENTS_PATH=""
+IMAGES_LENGTH=0
+
+# Functions
+
+checkInputs() {
+  ([ $1 != "null" ] && [ $2 != "null" ]) || die "Usage: sh iconizer.sh <IMAGE_PATH.pdf> <APP_ICON_PATH.appiconset>"
+  ([ -e "$1" ] && [ ${1: -4} == ".pdf" ]) || die "Did not find file $1, expected path to an image file with .pdf extension."
+  ([ -e "$2" ] && [ ${2: -11} == ".appiconset" ]) || die "Did not find folder $2, expected path should end with .appiconset extension."
+
+  PDF_PATH=$1
+  APP_ICON_SET_PATH=$2
+  CONTENTS_PATH="$APP_ICON_SET_PATH/Contents.json"
+
+  [ -e "$CONTENTS_PATH" ] || die "Did not find $CONTENTS_PATH, expected folder which contains $CONTENTS_PATH"
 
-[ $# == 2 ] || die "Usage: sh iconizer.sh <file name> <path to *.appiconset>"
-[ -e "$1" ] || die "Did not find file $1, expected path to an image file."
+  IMAGES_LENGTH=$(jq ".images | length" $CONTENTS_PATH)
 
-CONTENT_FILE_PATH="$2/$CONTENTS_FILE"
+  ([ $IMAGES_LENGTH != "null" ] && [ $IMAGES_LENGTH ] >0) || die "Did not find any images in $CONTENTS_PATH"
+}
+
+generateImages() {
+  echo "Creating icons from $APP_ICON_SET_PATH and updating $CONTENTS_PATH..."
+  for i in $(seq 0 $IMAGES_LENGTH); do
+    generateImage $i
+  done
+  echo "All Done 🎉🎉🎉"
+}
 
-[ -e "$CONTENT_FILE_PATH" ] || die "Did not find $CONTENT_FILE_PATH, expected folder which contains $CONTENTS_FILE"
+generateImage() {
+  image=$(jq ".images[$1]" $CONTENTS_PATH)
+  [ "$image" != "null" ] || break
 
-echo "Creating icons from $1 and updating $CONTENT_FILE_PATH..."
+  scale=$(echo $image | jq ".scale" | cut -d "\"" -f 2 | cut -d "x" -f 1)
```

Comment:
Removing the quotation marks is even easier:

```suggestion
  scale=$(echo $image | jq -r ".scale" | cut -d "x" -f 1)
```